low-level/files/opt/seagate/sspl/setup/sspl_setup.py

```diff
@@ -314,11 +316,12 @@ def validate(self):
             raise SetupError(1, msg)
         logger.info("%s - Validation done" % self.name)
 
-        if self.args.coverage and 'self' in self.args.plan[0]:
-            raise SetupError(errno.EINVAL,
-                             "%s - Argument validation failure. %s",
-                             self.name,
-                             "Code coverage can not be enabled with self tests.")
+        if self.args.coverage and self.args.plan[0] in IVT_TEST_PLANS:
```

Comment:
Wouldnt this change would disable coverage for all supported test plans today, it should not happen with sanity (self name earlier) test plan

Reply:
Coverage required service restart, and IVT_TEST_PLANS included only those test plans which are intended to run on HW and for which service restart or config change is not required.

If coverage is enabled with IVT test or self_primary or self_secondary plan it will raise an error.